core/include/edm/container.hpp

```diff
@@ -162,7 +162,7 @@ class container {
     /**
      * @brief Default Constructor
      */
-    TRACCC_HOST container() = default;
+    container() = default;
```

Comment:
This looks like an accidentally included change, or is this intended?

Reply:
I don't know if it was accidental or not, but it's definitely the right way to go! Not for CUDA, but rather for SYCL.

In vecmem I had to remove all host/device specifiers from the default constructors of the "data types" to make them work with oneAPI/SYCL.
  - https://github.com/acts-project/vecmem/blob/main/core/include/vecmem/containers/data/vector_view.hpp#L49
  - https://github.com/acts-project/vecmem/blob/main/core/include/vecmem/containers/data/jagged_vector_view.hpp#L56

It does mean that one can't use the default constructor of such types in device code, but the default constructor of `container` was not available in device code to begin with anyway. But this way at least it may be possible to pass such objects to a SYCL kernel.

Though most probably this was **not** Beomki's thinking behind this change...

Reply:
I removed the qualifier since there were warnings from it:
```
/home/beomki/projects/traccc/traccc/core/include/edm/container.hpp(165): warning: __host__ annotation is ignored on a function("container") that is explicitly defaulted on its first declaration
```
I didn't think that far ahead as atilla explained - but anyway it is nice if it can help SYCL

Reply:
Great, in that case we can just leave it! :smile: